src/movement_controls.py

```diff
@@ -0,0 +1,170 @@
+                    )
+        # the current acceleration is used to have speed progress from zero to its maximum
+        self.current_acceleration = self.clamp(
+            self.current_acceleration + (acceleration_change * delta_time), -1, 1
+        )
+
+        car_angle = math.radians(vehicle.angle)
+
+        # find the X and Y movement based on the angle of the sprite
+        self.current_velocity_x = (
+            self.drive_speed
+            * math.cos(car_angle)
+            * self.current_acceleration
+            * delta_time
+        )
+        self.current_velocity_y = (
+            self.drive_speed
+            * math.sin(car_angle)
+            * self.current_acceleration
+            * delta_time
+        )
+
+        # a stationary car cant turn
+        self.current_velocity_turn = (
+            input.x_axis.value
+            * self.turn_speed
+            * delta_time
+            * -self.current_acceleration
+        )
+
+    # called from the player to tell the vehicle to act on it's intended velocity and rotation
+    def move(self, delta_time: float, vehicle: arcade.Sprite, walls: arcade.SpriteList):
+        # the shadow sprite is used to simply math and planning to deal with the arena not being an array
+        self.shadow_sprite.center_x = vehicle.center_x + self.current_velocity_x
+        self.shadow_sprite.center_y = vehicle.center_y + self.current_velocity_y
+        self.shadow_sprite.angle = (
+            self.current_velocity_turn + vehicle.angle + self.external_velocity_turn
+        )
+        walls_touching_player = arcade.check_for_collision_with_list(
+            self.shadow_sprite, walls
+        )
+        if len(walls_touching_player) > 0:
+            # high speed impact should damage the car, and greatly cut the velocity
+            if self.current_acceleration > IMPACT_FORCE_FLOOR:
+                self.current_velocity_x -= self.current_velocity_x * 0.5
+                self.current_velocity_y -= self.current_velocity_y * 0.5
+            else:
+                self.current_acceleration -= (
+                    self.current_acceleration * self.acceleration_rate * delta_time
+                )
+                self.current_velocity_x = 0
+                self.current_velocity_y = 0
+                if abs(self.current_acceleration) < 0.1:
+                    self.current_acceleration = 0
+                    self.current_velocity_x = 0
+                    self.current_velocity_y = 0
+
+            self.shadow_sprite.center_x = vehicle.center_x
+            self.shadow_sprite.center_y = vehicle.center_y
+        else:
+            # no wall collisions means a valid spot to move to
+            vehicle.center_x = self.shadow_sprite.center_x
+            vehicle.center_y = self.shadow_sprite.center_y
+
+            vehicle.angle = self.shadow_sprite.angle
+        # NOTE: place holder boundaries to wrap aroung like pacman
```

Comment:
does having placeholder boundaries mean it doesn't wrap around? when testing this branch the car does not wrap around like pacman for me

Reply:
It doesn't wrap because it is colliding with the walls. If there weren't walls around the outside it would wrap around